src/Symfony/Component/HttpKernel/DataCollector/RequestDataCollector.php

```diff
@@ -62,7 +62,7 @@ public function collect(Request $request, Response $response, ?\Throwable $excep
         $sessionMetadata = [];
         $sessionAttributes = [];
         $flashes = [];
-        if (!$request->attributes->getBoolean('_stateless') && $request->hasSession()) {
+        if (!$request->isStateless() && $request->hasSession()) {
```

Comment:
We need to bump the version requirement for the HttpFoundation to `^7.2` for the HttpKernel component then.

Reply:
Isn't it the purpose of this change https://github.com/symfony/symfony/pull/57852/files#diff-d4302c4b8212e1bf33e8b1b2230034afc747cd93f565780a82c7df00a49155a9 ?

Reply:
ah sorry, I somehow missed that change